### PR TITLE
Fixes for missing login and layout

### PR DESCRIPTION
**Pull request by aforge**
Set auto login in SignupVC and CredentialsVC.
Fix message height calculation in MessageVC.
Route back to login screen in reset password VC.

### CODE REVIEW

File: Tinodios/CredentialsViewController.swift [hunk 1]
```diff
@@ -64,6 +64,9 @@ class CredentialsViewController : UIViewController {
                             UiUtils.showToast(message: "Verification failure: \(ctrl.code) \(ctrl.text)")
                         }
                     } else {
+                        if let token = tinode.authToken {
+                            tinode.setAutoLoginWithToken(token: token)
```
Comment (or-else) [inline Tinodios/CredentialsViewController.swift]: Why was it removed?

File: Tinodios/MessageViewController.swift [hunk 2]
```diff
@@ -964,7 +964,16 @@ extension MessageViewController : MessageViewLayoutDelegate {
             attributedText.append(NSAttributedString(string: "none", attributes: [.font: Constants.kContentFont]))
         }
         attributedText.append(NSAttributedString(string: carveout, attributes: [.font: Constants.kContentFont]))
-        let size = attributedText.boundingRect(with: CGSize(width: maxWidth, height: .greatestFiniteMagnitude), options: [.usesLineFragmentOrigin, .usesFontLeading], context: nil).integral.size
+
+        // Calculate content size.
+        // Per https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TextLayout/Tasks/StringHeight.html
+        let ts = NSTextStorage(attributedString: attributedText)
+        let tc = NSTextContainer(size: CGSize(width: maxWidth, height: .greatestFiniteMagnitude))
```
Comment (or-else) [inline Tinodios/MessageViewController.swift]: Can it be reused?

File: Tinodios/MessageViewController.swift [hunk 3]
```diff
@@ -964,7 +964,16 @@ extension MessageViewController : MessageViewLayoutDelegate {
...
+        let lm = NSLayoutManager()
```
Comment (or-else) [inline Tinodios/MessageViewController.swift]: ditto - can it be reused?

File: Tinodios/MessageViewController.swift [hunk 4]
```diff
@@ -964,7 +964,16 @@ extension MessageViewController : MessageViewLayoutDelegate {
...
+        lm.addTextContainer(tc)
+        ts.addLayoutManager(lm)
+
+        let size = lm.usedRect(for: tc).integral.size
```
Comment (or-else) [inline Tinodios/MessageViewController.swift]: why allocate size? Why not just `return`?

File: Tinodios/MessageViewController.swift (hunk 2 above)
Comment (aforge) [reply Tinodios/MessageViewController.swift]: Refactored in to UiUtils.TextSizeHelper.

File: Tinodios/MessageViewController.swift (hunk 4 above)
Comment (aforge) [reply Tinodios/MessageViewController.swift]: Yes.

File: Tinodios/MessageViewController.swift (hunk 3 above)
Comment (aforge) [reply Tinodios/MessageViewController.swift]: Done.

File: Tinodios/CredentialsViewController.swift (hunk 1 above)
Comment (aforge) [reply Tinodios/CredentialsViewController.swift]: We never set it. Looking at the existing code, auto login gets set separately in the app.
Both iOS and Android. We may move it to the SDK, though (loginSuccessful).

**Pull request merged**